tests/integration/suite/test_pod_security_policies.py

```diff
@@ -281,3 +281,65 @@ def check_project():
     wait_for(check_project)
 
     set_cluster_psp(admin_mc, "false")
+
+
+def test_psp_annotations(admin_mc, remove_resouce_func):
+    """Test that a psp with a pspt owner annotation will get cleaned up if the
+    parent pspt does not exist"""
+    k8s_client = kubernetes_api_client(admin_mc.client, 'local')
+    policy = kubernetes.client.PolicyV1beta1Api(api_client=k8s_client)
+    kubernetes.client.PolicyV1beta1PodSecurityPolicy
+    psp_name = random_str()
+    args = {
+        'metadata': {
+            'name': psp_name
+        },
+        'spec': {
+            "allowPrivilegeEscalation": True,
+            "fsGroup": {
+                "rule": "RunAsAny"
+            },
+            "runAsUser": {
+                "rule": "RunAsAny"
+            },
+            "seLinux": {
+                "rule": "RunAsAny"
+            },
+            "supplementalGroups": {
+                "rule": "RunAsAny"
+            },
+            "volumes": [
+                "*"
+            ]
+        }
+    }
+
+    psp = policy.create_pod_security_policy(args)
+    remove_resouce_func(policy.delete_pod_security_policy, psp_name)
+    psp = policy.read_pod_security_policy(psp_name)
+    assert psp is not None
```

Comment:
wouldn't it raise an ApiExcept if it couldn't find it? Also, should you retry this for some duration as even if it did delete it may not happen immediately right?

Reply:
The issue with waiting is how long is long enough? The patch will fail if the psp is gone